src/map/status.cpp

```diff
@@ -1000,8 +1000,15 @@ int status_damage(struct block_list *src,struct block_list *target,int64 dhp, in
 			for (const auto &it : status_db) {
 				sc_type type = static_cast<sc_type>(it.first);
 
-				if (sc->data[type] && it.second->flag[SCF_REMOVEONDAMAGED])
+				if (sc->data[type] && it.second->flag[SCF_REMOVEONDAMAGED]) {
+					if (sc->opt1 > OPT1_NONE) {
+						unit_data *ud = unit_bl2ud(target);
+
+						if (ud != nullptr)
+							ud->lastEffect = type;
```

Comment:
The name "lastEffect" is not very clear in what it does. We should at least put a comment like Skotlex did.
e.g.
// A status change that gets broken by damage should still be considered when calculating if a status change can be applied or not (for the same attack)
// This is a temporary solution until we refactor the code so that the calculation of an SC is done at the start of an attack rather than after the damage was applied